pkg/agent/openflow/client.go

```diff
@@ -1136,6 +1166,12 @@ func (c *client) SendUDPPacketOut(
 func (c *client) InstallMulticastInitialFlows(pktInReason uint8) error {
 	flows := c.featureMulticast.igmpPktInFlows(pktInReason)
 	flows = append(flows, c.featureMulticast.externalMulticastReceiverFlow())
+	// This flow is installed to bypass IGMP query statistics,
+	// making the remaining flows in this table only match non-IGMP ingress multicast traffic.
+	flows = append(flows, c.featureMulticast.multicastIngressIGMPPodMetricFlow())
+	// This flow is installed to bypass IGMP report statistics,
+	// making the remaining flows in this table only match non-IGMP egress multicast traffic.
+	flows = append(flows, c.featureMulticast.multicastEgressIGMPPodMetricFlow())
```

Comment:
I am thinking to use a single function to return two flows directly. Since the IGMP bypass flow is installed in MulticastXgressPodMetricTable which is after the XgressMetric flows ( used to collect statistics for IGMP packets ), we could not set dst IP address in the flow, just bypass all IGMP packets. What do you think @ceclinux ?

Reply:
Updated